### DISCUSSION TITLE
`find` node algorithm complexity

### DISCUSSION THREAD

**Discussion by lanyusan** [konvajs/konva#1218] [Q&A] [answered]
Hi, I am further evaluating Konva for my project.So far I am very happy with it. 

I am thinking of extensively using the `stage.find()` function. Can I check what's the complexity of this function of finding  nodes, especially `find` by `id`?

Is it random access of  O(1) or close to constant or linear O(n) or worse? This is a critical information for my evaluation.

Thanks in advance.

**Comment by lavrton** [comment, marked as answer]
I think it is O(n). It is not so hard to make a quick performance demo if it is important for you.

**Reply by lanyusan** [reply]
Thanks. Would do performance testing some time later, not only for this. But it is always good to hear from developer first of course.
